src/worker.cpp

```diff
@@ -138,15 +153,31 @@ void Worker::performInitialSetup()
     {
         if (typeid(ex) == std::type_index(typeid(DataException)))
         {
-            // TODO:Catch logic to be implemented once PEL code goes in.
+            EventLogger::createAsyncPelWithInventoryCallout(
+                types::ErrorType::InvalidVpdMessage,
+                types::SeverityType::Warning,
+                {{jsonUtility::getInventoryObjPathFromJson(
+                      m_parsedJson, SYSTEM_VPD_FILE_PATH),
+                  types::CalloutPriority::High}},
+                __FILE__, __FUNCTION__, 0, ex.what(), std::nullopt,
```

Comment:
should we add ?
"performInitialSetup failed , reason: " + ex.what() 
and is all below

Reply:
Done